README.md

```diff
@@ -7,43 +7,130 @@
+
+| Error                | IActionResult         | HTTP Status    |
+| -------------------- | --------------------- | -------------- |
+| `Error`*             | `BadRequestResult`    | 400 BadRequest |
+| `UnauthorizedError`  | `ForbidResult`        | 403 Forbidden  |
+| `NotFoundError`      | `NoContentResult`     | 404 NotFound   |
+
+_*This includes any other types that inherit from `Error` and are not explicitly listed._
+
+The defaults can be overriden by calling the extension method that takes an error mapping function. 
+This is useful when the domain model has defined additional error types and these need to be converted to the relevant problem details. 
+The status code that is set on the `ProblemDetails` will also be set on the `IActionResult` by the extension method so that the HTTP status code on the response is correct.
+
+For example consider a domain model that deals with payments. 
+It could be a news service which requires a subscription to access content. 
+It might contain several operations that require payment to be made before they can proceed. 
+This domain may therefore define a new error type as follows:
 
 ```csharp
-internal static class MvcConfigurer
+public class PaymentRequired : Error
 {
-    public static void ConfigureMvc(this MvcOptions options)
+    public PaymentRequired(string detail)
+        : base("Payment Required", detail)
     {
-        ...
-        options.Filters.Add(new DomainExceptionFilter());
     }
 }
 ```
 
-#### Extensibility
+It would therefore make sense to map this to a `402 Payment Required` HTTP response with relevant `ProblemDetails`. 
+This can be achieved like so:
+
+```csharp
+[HttpGet("{id}")]
+public async Task<IActionResult> GetNewsItem(string id)
+{
+    return await GetNewsItem(id)
+        .ToActionResult(
+            error => new ProblemDetails
+            {
+                Detail = error.Detail,
+                Status = StatusCodes.Status402PaymentRequired,
+                Title = error.Title,
+                Type = "https://example.com/problems/payment-required"
+            }
+        )
+}
+```
+
+The type field should return a URI that resolves to human-readable documentation about the type of error that has occurred. 
+This can either be existing documentation, such as https://httpstatuses.com for common errors, or your own documentation for domain-specific errors.
+
+Problem details is formally documented in [RFC 7807](https://tools.ietf.org/html/rfc7807). 
+More information about how the fields should be used can be found there.
 
-Since `DomainException` is extensible for any domain-specific error, `DomainExceptionFilter` can be extended to support custom Exception to Result mappings. Simply pass a `Func<DomainException, ErrorResponse, IActionResult>` to the constructor, such as
+It is likely that every action method in the API is going to need to perform this error mapping. 
+This is to maintain a loose coupling between API layer and domain model. 
```

Comment:
Feels a little clumsy here, something like this might be better:

`Action methods in the API typically perform this error mapping to maintain loose coupling between the API layer and domain model`